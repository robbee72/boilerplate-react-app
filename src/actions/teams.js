import uuid from 'uuid';
import database from '../firebase/firebase';

// ADD_TEAM
export const addTeam = (team) => ({
  type: 'ADD_TEAM',
  team
});

export const startAddTeam = (teamData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const {
      homeTown = '',
      teamName = '',
      note = '',
      conference = '',
      division = '',
      createdAt = 0
    } = teamData;
    const team = { homeTown, teamName, note, conference, division, createdAt };

    return database.ref(`users/${uid}/teams`).push(team).then((ref) => {
      dispatch(addTeam({
        id: ref.key,
        ...team
      }));
    });
  };
};

// REMOVE_TEAM
export const removeTeam = ({ id } = {}) => ({
  type: 'REMOVE_TEAM',
  id
});

export const startRemoveTeam = ({ id } = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/teams/${id}`).remove().then(() => {
      dispatch(removeTeam({ id }));
    });
  };
};

// EDIT_TEAM
export const editTeam = (id, updates) => ({
  type: 'EDIT_TEAM',
  id,
  updates
});

export const startEditTeam = (id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/teams/${id}`).update(updates).then(() => {
      dispatch(editTeam(id, updates));
    });
  };
};

// SET_TEAMS
export const setTeams = (teams) => ({
  type: 'SET_TEAMS',
  teams
});

export const startSetTeams = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/teams`).once('value').then((snapshot) => {
      const teams = [];

      snapshot.forEach((childSnapshot) => {
        teams.push({
          id: childSnapshot.key,
          ...childSnapshot.val()
        });
      });

      dispatch(setTeams(teams));
    });
  };
};
