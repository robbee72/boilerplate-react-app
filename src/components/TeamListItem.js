import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';


const TeamListItem = ({ id, homeTown, teamName, conference, division, createdAt }) => (
  <Link className="list-item" to={`/edit/${id}`}>
    <div>
      <h4>{homeTown} {teamName}   {conference} {division}</h4>

    <span>{moment(createdAt).format('MMMM Do, YYYY')}</span>
  </div>
  </Link>
);

export default TeamListItem;
