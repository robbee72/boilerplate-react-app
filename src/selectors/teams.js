import moment from 'moment';

// Get visible teams

export default (teams, { text, sortBy, startDate, endDate }) => {
  return teams.filter((team) => {
    const createdAtMoment = moment(team.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
    const textMatchHomeTown = team.homeTown.toLowerCase().includes(text.toLowerCase());
    const textMatchTeamName = team.teamName.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatchHomeTown && textMatchTeamName;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else if (sortBy === 'division') {
      return a.division < b.division ? 1 : -1;
    }
  });
};
