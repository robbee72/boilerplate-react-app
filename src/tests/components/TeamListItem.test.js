import React from 'react';
import { shallow } from 'enzyme';
import teams from '../fixtures/teams';
import TeamListItem from '../../components/TeamListItem';

test('should render TeamListItem correctly', () => {
  const wrapper = shallow(<TeamListItem {...teams[0]} />);
  expect(wrapper).toMatchSnapshot();
});
