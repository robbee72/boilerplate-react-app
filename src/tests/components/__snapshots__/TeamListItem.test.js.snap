// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TeamListItem correctly 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/1"
>
  <div>
    <h2>
      Vegas
    </h2>
    <h3>
      Golden Knights
    </h3>
    <span>
      December 31st, 1969
    </span>
  </div>
  <h3>
    $4.95
  </h3>
</Link>
`;
